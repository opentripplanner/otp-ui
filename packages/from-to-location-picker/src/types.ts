// eslint-disable-next-line prettier/prettier
import type { Location } from "@opentripplanner/types";

export type FromToPickerProps = {
  /**
   * Specifies the label to be rendered, or if set to true, renders the default label "Plan a trip:".
   */
  label?: boolean | React.ReactElement | string;
  /**
   * A specific location to associate with this. This is only used when combined
   * with the setLocation prop.
   */
  location?: Location;
  /**
   * Triggered when the user clicks on the "from" button.
   */
  onFromClick?: () => void;
  /**
   * Triggered when the user clicks on the "to" button.
   */
  onToClick?: () => void;
  /**
   * Triggered when the user clicks either the "from" or "to" button and there
   * are no from/to specific handler functions defined as props.
   *
   * Passes an argument as follows:
   * { locationType: "from/to", location, reverseGeocode: false }
   */
  setLocation?: (_: {
    locationType: string,
    // eslint-disable-next-line @typescript-eslint/no-shadow
    location: Location,
    reverseGeocode: boolean
  }) => void;
  /**
   * Determines whether icons are shown on the "from" and "to" buttons.
   */
  showIcons?: boolean;
};
