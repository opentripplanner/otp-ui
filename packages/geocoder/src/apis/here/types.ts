// Prettier does not support typescript annotation
// eslint-disable-next-line prettier/prettier
import type { LonLatInput } from "@conveyal/lonlat"

export interface HereResponse {
  items: Item[];
  point?: LonLatInput; // used in reverse response
}

export type Rect = {
  maxLat: number;
  maxLon: number;
  minLat: number;
  minLon: number;
};

export type Boundary = {
  country?: string;
  rect?: Rect;
};

export interface Item {
  access: Position[];
  address: Address;
  categories: Category[];
  distance: number;
  foodTypes: Category[];
  id: string;
  position: Position;
  resultType: string;
  scoring: Scoring;
  title: string;
}

export interface Position {
  lat: number;
  lng: number;
}

export interface Address {
  city: string;
  countryCode: string;
  countryName: string;
  county: string;
  district: string;
  houseNumber: string;
  label: string;
  postalCode: string;
  state: string;
  stateCode: string;
  street: string;
}

export interface Category {
  id: string;
  name: string;
  primary?: boolean;
}

export interface Scoring {
  fieldScore: FieldScore;
  queryScore: number;
}

export interface FieldScore {
  district: number;
  placeName: number;
}
