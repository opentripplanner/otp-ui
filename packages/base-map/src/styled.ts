import { Popup as MapGlPopup } from "react-map-gl";
import styled from "styled-components";

/**
 * Adds a box shadow and tweaks border radius to make popups easier to read.
 */
export const Popup = styled(MapGlPopup)`
  & > .maplibregl-popup-content,
  & > .mapboxgl-popup-content {
    border-radius: 10px;
    box-shadow: 0 3px 14px 4px rgb(0 0 0 / 20%);
  }
`;

export const MapOverlayPopup = styled.div`
  font-size: 12px;
  line-height: 1.5;
  min-width: 250px;
`;

export const PopupRow = styled.p`
  margin-top: 6px;
`;

export const PopupTitle = styled.header`
  font-size: 18px;
  font-weight: 500;
  margin-bottom: 6px;
`;

type LeafletStyleMarkerProps = {
  color?: string;
  size?: number;
  stroke?: number;
  strokeColor?: string;
};
/**
 * @deprecated this marker was created to make the transition from Leaflet more manageable,
 * but in most cases this marker should not be used -- use a MapLibreGL Circle instead
 *
 * https://maplibre.org/maplibre-gl-js-docs/style-spec/layers/#circle
 */
export const LeafletStyleMarker = styled.div<LeafletStyleMarkerProps>`
  background: ${props => props?.color || "#0000ff"}50;
  border-radius: ${props => props?.size || 10}px;
  border: ${props => props?.stroke || 2}px solid
    ${props => props?.strokeColor || `${props?.color}f0` || "#0000fffo"};
  content: "";
  cursor: pointer;
  display: block;
  height: ${props => props?.size || 10}px;
  width: ${props => props?.size || 10}px;
`;

export const LayerSelector = styled.aside`
  display: flex;
  justify-content: end;
  margin: 10px;
  position: relative;
  right: 0;
  top: 0;

  /* 
  There are some situations when using MapLibreGL's \`reuseMap\` where 
  a map re-render can cause the layer selector to be rendered multiple times. 
  This is a bit of a hack, but the most all-encompassing way to ensure that this is
  not a usability issue.

  This hack *only* works if the LayerSelector is *not* a \`div\`. I'm not sure why...
  */
  &:not(:last-of-type) {
    display: none;
  }

  .layers-list {
    background: rgba(255, 255, 255, 0.95);
    list-style-type: none;
    padding: 1em;
    position: absolute;
    right: 0;
    z-index: 1000;

    label {
      display: block;
      height: 0;
      overflow: hidden;
      width: 0;

      input {
        margin-right: 1ch;
      }
    }

    &::after {
      display: block;
      content: "üåê";
      cursor: pointer;
    }

    &:hover,
    &.fake-mobile-hover {
      box-shadow: 0px -1px 15px -3px rgba(0, 0, 0, 0.1);
      label {
        height: unset;
        overflow: unset;
        width: unset;
      }
      li:first-child {
        /* Prevent most accidental toggling of the first selectable layer on touch screens.
           Toggling is still possible if you touch the left-most area of the globe "button". */
        margin-right: 2em;
      }
      &::after {
        display: none;
      }
    }
  }
`;

/**
 * Map container for use with storybook across various packages in this repo.
 */
export const StoryMapContainer = styled.div`
  height: 90vh;
`;
