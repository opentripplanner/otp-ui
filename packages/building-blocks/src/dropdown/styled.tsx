import styled from "styled-components";
import grey from "../colors/grey";

export const DropdownButton = styled.button`
  background: #fff;
  border: 1px solid black;
  border-radius: 5px;
  color: inherit;
  padding: 5px 7px;
  transition: all 0.1s ease-in-out;

  span.caret {
    border-left: 4px solid transparent;
    border-right: 4px solid transparent;
    border-top: 4px solid;
    color: inherit;
    display: inline-block;
    height: 0;
    margin-left: 5px;
    vertical-align: middle;
    width: 0;
  }

  &:hover,
  &[aria-expanded="true"] {
    background: ${grey[50]};
    color: black;
    cursor: pointer;
  }
`;

export const DropdownMenu = styled.div<{ alignLeft?: boolean }>`
  background-clip: padding-box;
  background-color: #fff;
  border-radius: 4px;
  border: 1px solid rgba(0, 0, 0, 0.15);
  box-shadow: 0 6px 12px rgba(0, 0, 0, 0.175);
  color: #333;
  list-style: none;
  margin: 2px 0 0;
  min-width: 160px;
  padding: 5px 0;
  position: absolute;
  ${props => (props.alignLeft ? "left: 0;" : "right: 0;")}
  top: 100%;
  width: 100%;
  z-index: 1000;

  hr {
    margin: 0;
    padding: 0;
  }
  a,
  button {
    background: transparent;
    border: none;
    cursor: pointer;
    padding: 5px 15px;
    text-align: start;
    width: 100%;

    &:hover {
      background: ${grey[50]};
    }
  }
`;

export const DropdownWrapper = styled.span<{ pullRight?: boolean }>`
  float: ${props => (props.pullRight ? "right" : "left")};
  position: relative;
`;
