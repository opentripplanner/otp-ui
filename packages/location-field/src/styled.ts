import styled, { css, keyframes } from "styled-components";
import { Spinner as FASpinner } from "@styled-icons/fa-solid/Spinner";

const hiddenCss = css`
  clip: rect(0, 0, 0, 0);
  display: inline-block;
  height: 0;
  overflow: hidden;
  width: 0;
`;

export const HiddenContent = styled.span`
  ${hiddenCss}
`;

export const BaseButton = styled.button`
  border: none;
  background: none;
`;

export const ClearButton = styled(BaseButton)`
  color: #888;
  cursor: pointer;
  width: 30px;
`;

export const ClearBoth = styled.div`
  clear: both;
`;

export const DropdownButton = styled(BaseButton)`
  width: 30px;
`;

export const MenuItemList = styled.ul.attrs({
  role: "listbox"
})`
  background-clip: padding-box;
  background-color: #fff;
  border-radius: 4px;
  border: 1px solid rgba(0, 0, 0, 0.15);
  box-shadow: 0 6px 12px rgba(0, 0, 0, 0.175);
  float: left;
  font-size: 14px;
  left: 0;
  list-style: none;
  margin: 2px 0 0;
  min-width: 160px;
  padding: 5px 0;
  position: absolute;
  text-align: left;
  top: 100%;
  /* this is an annoyingly high number, but is needed to be on top of some otp-rr components */
  z-index: 1000000;

  /* If the associated input is not in an expanded state, hide the list. */
  input[aria-expanded="false"] ~ & {
    ${hiddenCss}
  }
`;

export const Input = styled.input`
  border: none;
  box-shadow: none;
  font-size: 17px;
  outline: none;
`;

export const InputGroup = styled.div`
  border-bottom: 1px solid #000;
  border-collapse: separate;
  display: table;
  margin-bottom: 15px;
  position: relative;
`;

export const MenuItemA = styled.a<{ active?: boolean }>`
  background-color: ${props => (props.active ? "#337ab7" : "transparent")};
  clear: both;
  color: ${props => (props.active ? "#fff" : "#333")};
  display: block;
  font-weight: 400;
  line-height: 1.42857143;
  padding: 3px 20px;
  text-decoration: none;
  white-space: nowrap;
`;

export const MenuGroupHeader = styled.h2<{
  bgColor?: string;
  fgColor?: string;
}>`
  color: ${props => props.fgColor || "#eee"};
  background-color: ${props => props.bgColor || "#333"};
  font-size: 12px;
  font-weight: normal;
  line-height: 1.42857143;
  margin: 0;
  padding: 0px 10px;
  text-align: center;
  white-space: nowrap;
`;

export const MenuItemLi = styled.li`
  &:hover {
    /* TODO: adjust highlight color based on props.color? */
    background-color: #f5f5f5;
    cursor: pointer;
  }

  /* For disabled (aria-hidden) elements,
     show an arrow pointer (no I-beam) and don't highlight the background. */
  &[aria-hidden="true"]:hover {
    background-color: unset;
    cursor: default;
  }
`;

export const OptionContainer = styled.span`
  display: block;
  padding-top: 5px;
  padding-bottom: 3px;
`;

export const OptionSubTitle = styled.span`
  color: #686868;
  font-size: 12px;
  margin-left: 6px;
`;

export const OptionContent = styled.span`
  margin-left: 30px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
`;

export const OptionIconContainer = styled.span`
  float: left;
`;

export const RouteName = styled.span`
  background-color: gray;
  color: white;
  padding: 2px 3px 0px;
  margin-right: 5px;
`;

export const StaticMenuItemList = styled(MenuItemList)`
  border: none;
  box-shadow: none;
  display: block;
`;

export const StopContentContainer = styled.span`
  margin-left: 30px;
`;

export const StopDistance = styled.span`
  font-size: 8px;
`;

export const StopIconAndDistanceContainer = styled.span`
  float: left;
  padding-top: 3px;
`;

export const StopName = styled.span``;

export const StopRoutes = styled.span`
  font-size: 9px;
`;

const rotateAnimation = keyframes`
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
`;

export const Spinner = styled(FASpinner)`
  animation: ${rotateAnimation} 1.2s linear infinite;
`;
