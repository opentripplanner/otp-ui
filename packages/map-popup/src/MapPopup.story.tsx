import React from "react";
import { action } from "@storybook/addon-actions";
import styled from "styled-components";
import { Station, Stop } from "@opentripplanner/types";
import { IntlProvider } from "react-intl";
import { Meta } from "@storybook/react";
import MapPopupContents from "./index";

// HOC to wrap components with IntlProvider
const withIntl = (Story: () => JSX.Element) => (
  <IntlProvider messages={{}} locale="en">
    <Story />
  </IntlProvider>
);

export default {
  title: "Map Popup",
  decorators: [withIntl]
} as Meta;

const STOP = {
  flex: false,
  gtfsId: "9526",
  id: "9526",
  lat: 45.523009,
  lon: -122.672529,
  name: "W Burnside & SW 2nd"
};

const STATION = {
  "stroke-width": 2,
  allowDropoff: true,
  allowPickup: true,
  bikesAvailable: 6,
  color: "#f00",
  id: '"hub_1580"',
  isCarStation: false,
  isFloatingBike: false,
  name: "SW Morrison at 18th",
  networks: ["BIKETOWN"],
  realTimeData: true,
  spacesAvailable: 11,
  x: -122.6896771788597,
  y: 45.5219604810172
};

const FLOATING_VEHICLE = {
  "stroke-width": 1,
  allowDropoff: false,
  allowPickup: true,
  bikesAvailable: 1,
  color: "#f00",
  id: '"bike_6861"',
  isCarStation: false,
  isFloatingBike: true,
  name: "0541",
  networks: ["BIKETOWN"],
  realTimeData: true,
  spacesAvailable: 0,
  x: -122.70486,
  y: 45.525486666666666
};

const FLOATING_CAR = {
  "stroke-width": 1,
  allowDropoff: false,
  allowPickup: true,
  color: "#333",
  id: "car_6861",
  isCarStation: false,
  isFloatingCar: true,
  name: "0541",
  networks: ["MILES"], // https://miles-mobility.com
  realTimeData: true,
  spacesAvailable: 0,
  x: 13.405,
  y: 52.52
};

const getEntityPrefixExample = (entity: Stop | Station) => {
  const DemoIcon = styled.span`
    background-color: blue;
    border-radius: 50px;
    color: white;
    margin-right: 0.5ch;
    padding: 2px;
  `;

  return <DemoIcon>{entity.name?.charAt(0)}</DemoIcon>;
};

export const StopEntity = (): JSX.Element => (
  <MapPopupContents
    entity={STOP}
    setLocation={action("setLocation")}
    setViewedStop={action("setViewedStop")}
  />
);
export const StopEntitywithEntityPrefix = (): JSX.Element => (
  <MapPopupContents
    entity={STOP}
    getEntityPrefix={getEntityPrefixExample}
    setLocation={action("setLocation")}
    setViewedStop={action("setViewedStop")}
  />
);

export const StopEntityNoHandlers = (): JSX.Element => (
  <MapPopupContents entity={STOP} />
);

export const StationEntity = (): JSX.Element => (
  <MapPopupContents
    entity={STATION}
    setLocation={action("setLocation")}
    setViewedStop={action("setViewedStop")}
  />
);

export const FloatingCarEntity = (): JSX.Element => (
  <MapPopupContents entity={FLOATING_CAR} setLocation={action("setLocation")} />
);

export const FloatingVehicleEntity = (): JSX.Element => (
  <MapPopupContents
    entity={FLOATING_VEHICLE}
    setLocation={action("setLocation")}
    setViewedStop={action("setViewedStop")}
  />
);
