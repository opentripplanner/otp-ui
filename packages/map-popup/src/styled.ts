import styled from "styled-components";

/* eslint-disable-next-line import/prefer-default-export */
export const ViewStopButton = styled.button<{ stopId?: string }>`
  background: none;
  border-bottom: none;
  ${props =>
    props.stopId
      ? "border-left: 1px solid #000; margin-left: 5px;"
      : "border-left: none; padding-left: 0;"};
  border-right: none;
  border-top: none;
  color: #008;
  font-family: inherit;
  padding-top: 0;
`;
