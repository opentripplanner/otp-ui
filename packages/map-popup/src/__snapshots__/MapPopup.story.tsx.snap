// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map Popup FloatingCarEntity smoke-test 1`] = `
<div class="styled__MapOverlayPopup-sc-12kjso7-1 cPqJxe">
  <div id="focus-car_6861-popup-focus-trap"
       role="presentation"
  >
    <header class="styled__PopupTitle-sc-12kjso7-3 jRNaQh">
      MILES Car: 0541
    </header>
    <p class="styled__PopupRow-sc-12kjso7-2 ckOOWr">
      <strong>
        Plan a trip:
      </strong>
      <span class="styled__FromToPickerSpan-sc-vb4790-1 giBPod">
        <span class="styled__LocationPickerSpan-sc-vb4790-0 gsVfXo">
          <svg viewbox="0 0 512 512"
               height="0.9em"
               width="0.9em"
               aria-hidden="true"
               focusable="false"
               fill="currentColor"
               xmlns="http://www.w3.org/2000/svg"
               class="StyledIconBase-sc-ea9ulj-0 ebjPRL styled__FromIcon-sc-n5xcvc-0 dDqEuj"
          >
            <path fill="currentColor"
                  d="M160 256c0-53.9 42.1-96 96-96 53 0 96 42.1 96 96 0 53-43 96-96 96-53.9 0-96-43-96-96zm352 0c0 141.4-114.6 256-256 256S0 397.4 0 256 114.6 0 256 0s256 114.6 256 256zM256 48C141.1 48 48 141.1 48 256s93.1 208 208 208 208-93.1 208-208S370.9 48 256 48z"
            >
            </path>
          </svg>
          <button class="styled__Button-sc-vb4790-2 ekklXB">
            From here
          </button>
        </span>
        <span class="styled__LocationPickerSpan-sc-vb4790-0 gsVfXo">
          <svg viewbox="0 0 384 512"
               height="0.9em"
               width="0.9em"
               aria-hidden="true"
               focusable="false"
               fill="currentColor"
               xmlns="http://www.w3.org/2000/svg"
               class="StyledIconBase-sc-ea9ulj-0 ebjPRL styled__ToIcon-sc-n5xcvc-2 gZxRwk"
          >
            <path fill="currentColor"
                  d="M215.7 499.2C267 435 384 279.4 384 192 384 86 298 0 192 0S0 86 0 192c0 87.4 117 243 168.3 307.2 12.3 15.3 35.1 15.3 47.4 0zM192 256c-35.3 0-64-28.7-64-64s28.7-64 64-64 64 28.7 64 64-28.7 64-64 64z"
            >
            </path>
          </svg>
          <button class="styled__Button-sc-vb4790-2 ekklXB">
            To here
          </button>
        </span>
      </span>
    </p>
  </div>
</div>
`;

exports[`Map Popup FloatingVehicleEntity smoke-test 1`] = `
<div class="styled__MapOverlayPopup-sc-12kjso7-1 cPqJxe">
  <div id="focus-22bike_686122-popup-focus-trap"
       role="presentation"
  >
    <header class="styled__PopupTitle-sc-12kjso7-3 jRNaQh">
      BIKETOWN Free-floating bike: 0541
    </header>
    <p class="styled__PopupRow-sc-12kjso7-2 ckOOWr">
      <strong>
        Plan a trip:
      </strong>
      <span class="styled__FromToPickerSpan-sc-vb4790-1 giBPod">
        <span class="styled__LocationPickerSpan-sc-vb4790-0 gsVfXo">
          <svg viewbox="0 0 512 512"
               height="0.9em"
               width="0.9em"
               aria-hidden="true"
               focusable="false"
               fill="currentColor"
               xmlns="http://www.w3.org/2000/svg"
               class="StyledIconBase-sc-ea9ulj-0 ebjPRL styled__FromIcon-sc-n5xcvc-0 dDqEuj"
          >
            <path fill="currentColor"
                  d="M160 256c0-53.9 42.1-96 96-96 53 0 96 42.1 96 96 0 53-43 96-96 96-53.9 0-96-43-96-96zm352 0c0 141.4-114.6 256-256 256S0 397.4 0 256 114.6 0 256 0s256 114.6 256 256zM256 48C141.1 48 48 141.1 48 256s93.1 208 208 208 208-93.1 208-208S370.9 48 256 48z"
            >
            </path>
          </svg>
          <button class="styled__Button-sc-vb4790-2 ekklXB">
            From here
          </button>
        </span>
        <span class="styled__LocationPickerSpan-sc-vb4790-0 gsVfXo">
          <svg viewbox="0 0 384 512"
               height="0.9em"
               width="0.9em"
               aria-hidden="true"
               focusable="false"
               fill="currentColor"
               xmlns="http://www.w3.org/2000/svg"
               class="StyledIconBase-sc-ea9ulj-0 ebjPRL styled__ToIcon-sc-n5xcvc-2 gZxRwk"
          >
            <path fill="currentColor"
                  d="M215.7 499.2C267 435 384 279.4 384 192 384 86 298 0 192 0S0 86 0 192c0 87.4 117 243 168.3 307.2 12.3 15.3 35.1 15.3 47.4 0zM192 256c-35.3 0-64-28.7-64-64s28.7-64 64-64 64 28.7 64 64-28.7 64-64 64z"
            >
            </path>
          </svg>
          <button class="styled__Button-sc-vb4790-2 ekklXB">
            To here
          </button>
        </span>
      </span>
    </p>
  </div>
</div>
`;

exports[`Map Popup StationEntity smoke-test 1`] = `
<div class="styled__MapOverlayPopup-sc-12kjso7-1 cPqJxe">
  <div id="focus-22hub_158022-popup-focus-trap"
       role="presentation"
  >
    <header class="styled__PopupTitle-sc-12kjso7-3 jRNaQh">
      BIKETOWN SW Morrison at 18th
    </header>
    <p class="styled__PopupRow-sc-12kjso7-2 ckOOWr">
      <div>
        Available bikes: 6
      </div>
      <div>
        Available docks: 11
      </div>
    </p>
    <p class="styled__PopupRow-sc-12kjso7-2 ckOOWr">
      <strong>
        Plan a trip:
      </strong>
      <span class="styled__FromToPickerSpan-sc-vb4790-1 giBPod">
        <span class="styled__LocationPickerSpan-sc-vb4790-0 gsVfXo">
          <svg viewbox="0 0 512 512"
               height="0.9em"
               width="0.9em"
               aria-hidden="true"
               focusable="false"
               fill="currentColor"
               xmlns="http://www.w3.org/2000/svg"
               class="StyledIconBase-sc-ea9ulj-0 ebjPRL styled__FromIcon-sc-n5xcvc-0 dDqEuj"
          >
            <path fill="currentColor"
                  d="M160 256c0-53.9 42.1-96 96-96 53 0 96 42.1 96 96 0 53-43 96-96 96-53.9 0-96-43-96-96zm352 0c0 141.4-114.6 256-256 256S0 397.4 0 256 114.6 0 256 0s256 114.6 256 256zM256 48C141.1 48 48 141.1 48 256s93.1 208 208 208 208-93.1 208-208S370.9 48 256 48z"
            >
            </path>
          </svg>
          <button class="styled__Button-sc-vb4790-2 ekklXB">
            From here
          </button>
        </span>
        <span class="styled__LocationPickerSpan-sc-vb4790-0 gsVfXo">
          <svg viewbox="0 0 384 512"
               height="0.9em"
               width="0.9em"
               aria-hidden="true"
               focusable="false"
               fill="currentColor"
               xmlns="http://www.w3.org/2000/svg"
               class="StyledIconBase-sc-ea9ulj-0 ebjPRL styled__ToIcon-sc-n5xcvc-2 gZxRwk"
          >
            <path fill="currentColor"
                  d="M215.7 499.2C267 435 384 279.4 384 192 384 86 298 0 192 0S0 86 0 192c0 87.4 117 243 168.3 307.2 12.3 15.3 35.1 15.3 47.4 0zM192 256c-35.3 0-64-28.7-64-64s28.7-64 64-64 64 28.7 64 64-28.7 64-64 64z"
            >
            </path>
          </svg>
          <button class="styled__Button-sc-vb4790-2 ekklXB">
            To here
          </button>
        </span>
      </span>
    </p>
  </div>
</div>
`;

exports[`Map Popup StopEntity smoke-test 1`] = `
<div class="styled__MapOverlayPopup-sc-12kjso7-1 cPqJxe">
  <div id="focus-9526-popup-focus-trap"
       role="presentation"
  >
    <header class="styled__PopupTitle-sc-12kjso7-3 jRNaQh">
      W Burnside &amp; SW 2nd
    </header>
    <p class="styled__PopupRow-sc-12kjso7-2 ckOOWr">
      <strong>
        Stop ID: 9526
      </strong>
      <button class="styled__ViewStopButton-sc-12v7ov3-0 hXaHvR">
        Stop Viewer
      </button>
    </p>
    <p class="styled__PopupRow-sc-12kjso7-2 ckOOWr">
      <strong>
        Plan a trip:
      </strong>
      <span class="styled__FromToPickerSpan-sc-vb4790-1 giBPod">
        <span class="styled__LocationPickerSpan-sc-vb4790-0 gsVfXo">
          <svg viewbox="0 0 512 512"
               height="0.9em"
               width="0.9em"
               aria-hidden="true"
               focusable="false"
               fill="currentColor"
               xmlns="http://www.w3.org/2000/svg"
               class="StyledIconBase-sc-ea9ulj-0 ebjPRL styled__FromIcon-sc-n5xcvc-0 dDqEuj"
          >
            <path fill="currentColor"
                  d="M160 256c0-53.9 42.1-96 96-96 53 0 96 42.1 96 96 0 53-43 96-96 96-53.9 0-96-43-96-96zm352 0c0 141.4-114.6 256-256 256S0 397.4 0 256 114.6 0 256 0s256 114.6 256 256zM256 48C141.1 48 48 141.1 48 256s93.1 208 208 208 208-93.1 208-208S370.9 48 256 48z"
            >
            </path>
          </svg>
          <button class="styled__Button-sc-vb4790-2 ekklXB">
            From here
          </button>
        </span>
        <span class="styled__LocationPickerSpan-sc-vb4790-0 gsVfXo">
          <svg viewbox="0 0 384 512"
               height="0.9em"
               width="0.9em"
               aria-hidden="true"
               focusable="false"
               fill="currentColor"
               xmlns="http://www.w3.org/2000/svg"
               class="StyledIconBase-sc-ea9ulj-0 ebjPRL styled__ToIcon-sc-n5xcvc-2 gZxRwk"
          >
            <path fill="currentColor"
                  d="M215.7 499.2C267 435 384 279.4 384 192 384 86 298 0 192 0S0 86 0 192c0 87.4 117 243 168.3 307.2 12.3 15.3 35.1 15.3 47.4 0zM192 256c-35.3 0-64-28.7-64-64s28.7-64 64-64 64 28.7 64 64-28.7 64-64 64z"
            >
            </path>
          </svg>
          <button class="styled__Button-sc-vb4790-2 ekklXB">
            To here
          </button>
        </span>
      </span>
    </p>
  </div>
</div>
`;

exports[`Map Popup StopEntityNoHandlers smoke-test 1`] = `
<div class="styled__MapOverlayPopup-sc-12kjso7-1 cPqJxe">
  <div id="focus-9526-popup-focus-trap"
       role="presentation"
  >
    <header class="styled__PopupTitle-sc-12kjso7-3 jRNaQh">
      W Burnside &amp; SW 2nd
    </header>
  </div>
</div>
`;

exports[`Map Popup StopEntityNoStopCode smoke-test 1`] = `
<div class="styled__MapOverlayPopup-sc-12kjso7-1 cPqJxe">
  <div id="focus-9526-popup-focus-trap"
       role="presentation"
  >
    <header class="styled__PopupTitle-sc-12kjso7-3 jRNaQh">
      W Burnside &amp; SW 2nd
    </header>
    <p class="styled__PopupRow-sc-12kjso7-2 ckOOWr">
      <button class="styled__ViewStopButton-sc-12v7ov3-0 bbtcwi">
        Stop Viewer
      </button>
    </p>
    <p class="styled__PopupRow-sc-12kjso7-2 ckOOWr">
      <strong>
        Plan a trip:
      </strong>
      <span class="styled__FromToPickerSpan-sc-vb4790-1 giBPod">
        <span class="styled__LocationPickerSpan-sc-vb4790-0 gsVfXo">
          <svg viewbox="0 0 512 512"
               height="0.9em"
               width="0.9em"
               aria-hidden="true"
               focusable="false"
               fill="currentColor"
               xmlns="http://www.w3.org/2000/svg"
               class="StyledIconBase-sc-ea9ulj-0 ebjPRL styled__FromIcon-sc-n5xcvc-0 dDqEuj"
          >
            <path fill="currentColor"
                  d="M160 256c0-53.9 42.1-96 96-96 53 0 96 42.1 96 96 0 53-43 96-96 96-53.9 0-96-43-96-96zm352 0c0 141.4-114.6 256-256 256S0 397.4 0 256 114.6 0 256 0s256 114.6 256 256zM256 48C141.1 48 48 141.1 48 256s93.1 208 208 208 208-93.1 208-208S370.9 48 256 48z"
            >
            </path>
          </svg>
          <button class="styled__Button-sc-vb4790-2 ekklXB">
            From here
          </button>
        </span>
        <span class="styled__LocationPickerSpan-sc-vb4790-0 gsVfXo">
          <svg viewbox="0 0 384 512"
               height="0.9em"
               width="0.9em"
               aria-hidden="true"
               focusable="false"
               fill="currentColor"
               xmlns="http://www.w3.org/2000/svg"
               class="StyledIconBase-sc-ea9ulj-0 ebjPRL styled__ToIcon-sc-n5xcvc-2 gZxRwk"
          >
            <path fill="currentColor"
                  d="M215.7 499.2C267 435 384 279.4 384 192 384 86 298 0 192 0S0 86 0 192c0 87.4 117 243 168.3 307.2 12.3 15.3 35.1 15.3 47.4 0zM192 256c-35.3 0-64-28.7-64-64s28.7-64 64-64 64 28.7 64 64-28.7 64-64 64z"
            >
            </path>
          </svg>
          <button class="styled__Button-sc-vb4790-2 ekklXB">
            To here
          </button>
        </span>
      </span>
    </p>
  </div>
</div>
`;

exports[`Map Popup StopEntityWithEntityPrefix smoke-test 1`] = `
<div class="styled__MapOverlayPopup-sc-12kjso7-1 cPqJxe">
  <div id="focus-9526-popup-focus-trap"
       role="presentation"
  >
    <header class="styled__PopupTitle-sc-12kjso7-3 jRNaQh">
      <span class="MapPopupstory__DemoIcon-sc-vlk9f0-0 hzuFzw">
        W
      </span>
      W Burnside &amp; SW 2nd
    </header>
    <p class="styled__PopupRow-sc-12kjso7-2 ckOOWr">
      <strong>
        Stop ID: 9526
      </strong>
      <button class="styled__ViewStopButton-sc-12v7ov3-0 hXaHvR">
        Stop Viewer
      </button>
    </p>
    <p class="styled__PopupRow-sc-12kjso7-2 ckOOWr">
      <strong>
        Plan a trip:
      </strong>
      <span class="styled__FromToPickerSpan-sc-vb4790-1 giBPod">
        <span class="styled__LocationPickerSpan-sc-vb4790-0 gsVfXo">
          <svg viewbox="0 0 512 512"
               height="0.9em"
               width="0.9em"
               aria-hidden="true"
               focusable="false"
               fill="currentColor"
               xmlns="http://www.w3.org/2000/svg"
               class="StyledIconBase-sc-ea9ulj-0 ebjPRL styled__FromIcon-sc-n5xcvc-0 dDqEuj"
          >
            <path fill="currentColor"
                  d="M160 256c0-53.9 42.1-96 96-96 53 0 96 42.1 96 96 0 53-43 96-96 96-53.9 0-96-43-96-96zm352 0c0 141.4-114.6 256-256 256S0 397.4 0 256 114.6 0 256 0s256 114.6 256 256zM256 48C141.1 48 48 141.1 48 256s93.1 208 208 208 208-93.1 208-208S370.9 48 256 48z"
            >
            </path>
          </svg>
          <button class="styled__Button-sc-vb4790-2 ekklXB">
            From here
          </button>
        </span>
        <span class="styled__LocationPickerSpan-sc-vb4790-0 gsVfXo">
          <svg viewbox="0 0 384 512"
               height="0.9em"
               width="0.9em"
               aria-hidden="true"
               focusable="false"
               fill="currentColor"
               xmlns="http://www.w3.org/2000/svg"
               class="StyledIconBase-sc-ea9ulj-0 ebjPRL styled__ToIcon-sc-n5xcvc-2 gZxRwk"
          >
            <path fill="currentColor"
                  d="M215.7 499.2C267 435 384 279.4 384 192 384 86 298 0 192 0S0 86 0 192c0 87.4 117 243 168.3 307.2 12.3 15.3 35.1 15.3 47.4 0zM192 256c-35.3 0-64-28.7-64-64s28.7-64 64-64 64 28.7 64 64-28.7 64-64 64z"
            >
            </path>
          </svg>
          <button class="styled__Button-sc-vb4790-2 ekklXB">
            To here
          </button>
        </span>
      </span>
    </p>
  </div>
</div>
`;
